korangar/src/graphics/passes/forward/shader/entity.wgsl

```diff
@@ -107,30 +105,30 @@ fn vs_main(
     let rotated = rotateY(vec3<f32>(global_uniforms.view[2].x, 0, global_uniforms.view[2].z), vertex.position.x);
     output.normal = vec3<f32>(-rotated.x, rotated.y, rotated.z);
 
+    let MAX_X = 500.0;
+    let MAX_Y = 500.0;
+
+    // Instead of using -1 to 1, use the proportion size.
+    let proportion_x = instance.texture_size.x/MAX_X;
+    let proportion_y = instance.texture_size.y/MAX_Y;
+
     // The depth multiplier and curvature multiplier is derived from the truth table of vertex_data
     // Because we have to transform the vertex of the frame part, we can't use the depth and curvature
     // directly and are using the fact, that y/depth and x/curvature correlate to each other.
     // An offset is also added for frame parts not stay at the same depth.
-    output.depth_offset = frame_part_vertex.y / 2.0 + instance.extra_depth_offset;
-    output.curvature = frame_part_vertex.x;
-
+    output.depth_offset = frame_part_vertex.y * proportion_y / 2.0 + instance.extra_depth_offset;
+    output.curvature = frame_part_vertex.x * proportion_x;
+    
     output.original_depth_offset = instance.depth_offset;
     output.original_curvature = instance.curvature;
-    output.angle = instance.angle;
     output.color = instance.color;
     return output;
 }
 
 @fragment
 fn fs_main(input: VertexOutput) -> FragmentOutput {
-    // Apply the rotation from action
-    let sin_factor = sin(input.angle);
-    let cos_factor = cos(input.angle);
-    let rotate = vec2(input.texture_coordinates.x - 0.5, input.texture_coordinates.y - 0.5) * mat2x2(cos_factor, sin_factor, -sin_factor, cos_factor);
-    let texture_coordinates = vec2(clamp(rotate.x + 0.5, 0.0, 1.0), clamp(rotate.y + 0.5, 0.0, 1.0));
-
-    let diffuse_color = textureSample(texture, texture_sampler, texture_coordinates);
-    let alpha_channel = textureSample(texture, nearest_sampler, texture_coordinates).a;
+    let diffuse_color = textureSample(texture, texture_sampler, input.texture_coordinates);
+    let alpha_channel = textureSample(texture, nearest_sampler, input.texture_coordinates).a;
 
     if (alpha_channel == 0.0) {
```

Comment:
Don't we need to also discard here if the input.color alpha is not 1.0?

Reply:
Yes, I will change there.